docs/ota-client-guide/modules/ROOT/pages/garage-sign-reference.adoc

```diff
@@ -0,0 +1,357 @@
+
++++<details><summary>+++
+`targets [init|add|add-uploaded|delete|sign|pull|push|upload|delegations]`—(Only for the repo server) Manages targets.
+Targets are software versions available in your OTA Connect software repository.
++++</summary><div>+++
+
++++<details><summary>+++
+`targets init`—Creates a `targets.json` file.
++++</summary><div>+++
+
+[.release_notes]
+[cols="15m,75a"]
+|====================
+| --version | The version of the `targets.json` file. The versions that start at 1 and always have to increase.
+| --expires | The expiry time of your target. It is a UTC instant, such as `2020-01-01T00:01:00Z`.
+|====================
++++</div></details>+++
+
++++<details><summary>+++
+`targets add`—Adds a target.
++++</summary><div>+++
+
+[.release_notes]
+[cols="15m,75a"]
+|====================
+| --length | The length of the target, in bytes.
+| --name | The name of the target.
+| --version | The versions of the target.
+| --format | The format of the target: [`ostree`\|`binary`]
+| --sha256 | The hash of the binary.
+| --hardwareids | The types of hardware for which your binary file is intended.
+| --url | The external URL where the `targets.json` file can be downloaded.
+|====================
++++</div></details>+++
+
++++<details><summary>+++
+`targets add-uploaded`—Adds the modified `targets.json` file.
++++</summary><div>+++
+
+[.release_notes]
+[cols="15m,75a"]
+|====================
+| -i, --input | The path to the binary file.
+| --name | The name of the target.
+| --version | The versions of the target.
+| --hardwareids | The types of hardware for which your binary file is intended.
+|====================
++++</div></details>+++
+
++++<details><summary>+++
+`targets delete`—Deletes a single target.
++++</summary><div>+++
+
+[.release_notes]
+[cols="15m,75a"]
+|====================
+| --filename | The name of the file with the target that you want to delete in the following format: <name>_<version>.
+|====================
++++</div></details>+++
+
++++<details><summary>+++
+`targets sign`—Signs your `targets.json` file.
++++</summary><div>+++
+
+[.release_notes]
+[cols="15m,75a"]
+|====================
+| --key-name | The name of the file with the target that you want to delete in the following format: <name>_<version>.
+|--version | The versions of the target. It ignores an unsigned role version and uses the specified version instead.
+| --expires | The expiry time of your target. It is a UTC instant, such as `2020-01-01T00:01:00Z`.
+| --expire-after | The expiration delay in years, months, and days (each optional, but in that order), such as `1Y3M5D`.
+| --force | Sets a date in the past.
+|====================
++++</div></details>+++
+
+`targets pull`—Pulls the current `targets.json` file from OTA Connect.
+
+`targets push`—Pushes the latest `targets.json` file to the server.
+If the latest `pull` was too long ago, and you did not pull the latest `targets.json` on the server, this `push` will fail with exit code 2.
```

Comment:
```suggestion
If the targets file is invalid, for example because of a bad signature or a non-increasing version number, this `push` will fail with exit code 2.
```